mmv1/third_party/terraform/resources/resource_dns_record_set.go

```diff
@@ -186,6 +221,81 @@ func resourceDnsRecordSet() *schema.Resource {
 	}
 }
 
+var geoPolicySchema *schema.Resource = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"location": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: `The location name defined in Google Cloud.`,
+		},
+		"rrdatas": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"health_checked_targets": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Description: "For A and AAAA types only. The list of targets to be health checked. These can be specified along with `rrdatas` within this item.",
+			MaxItems:    1,
+			Elem:        healthCheckedTargetSchema,
+		},
+	},
+}
+
+var healthCheckedTargetSchema *schema.Resource = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"internal_load_balancers": {
+			Type:        schema.TypeList,
+			Required:    true,
+			Description: "The list of internal load balancers to health check.",
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"load_balancer_type": {
+						Type:         schema.TypeString,
+						Required:     true,
+						Description:  `The type of load balancer. This value is case-sensitive. Possible values: ["regionalL4ilb"]`,
+						ValidateFunc: validation.StringInSlice([]string{"regionalL4ilb"}, false),
+					},
+					"ip_address": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "The frontend IP address of the load balancer.",
+					},
+					"port": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "The configured port of the load balancer.",
+					},
+					"ip_protocol": {
+						Type:         schema.TypeString,
+						Required:     true,
+						Description:  `The configured IP protocol of the load balancer. This value is case-sensitive. Possible values: ["tcp", "udp"]`,
+						ValidateFunc: validation.StringInSlice([]string{"tcp", "udp"}, false),
+					},
+					"network_url": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "The fully qualified url of the network in which the load balancer belongs. This should be formatted like `https://www.googleapis.com/compute/v1/projects/{project}/global/networks/{network}`.",
+					},
+					"project": {
+						Type:        schema.TypeString,
+						Required:    true,
+						Description: "The ID of the project in which the load balancer belongs.",
+					},
+					"region": {
+						Type:        schema.TypeString,
+						Optional:    true,
```

Comment:
I was not able to come up with a test scenario with an internal load balancer that was global, so I did not confirm whether `region` was required in this case. I've left it `Optional` because I assume that it is not needed for global load balancers.